core/server/common/src/main/java/alluxio/master/audit/AsyncUserAccessAuditLogWriter.java

```diff
@@ -0,0 +1,83 @@
+package alluxio.master.audit;
+
+import alluxio.Configuration;
+import alluxio.PropertyKey;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.concurrent.ThreadSafe;
+
+@ThreadSafe
+public final class AsyncUserAccessAuditLogWriter {
+  private static final int QUEUE_SIZE = 10000;
+  private static final int BLOCKING_TIMEOUT_US = 10;
+  private static final Logger LOG = LoggerFactory.getLogger(AsyncUserAccessAuditLogWriter.class);
+  private boolean mEnabled;
+  private volatile boolean mStopped;
+  private ArrayBlockingQueue<AuditContext> mAuditLogEntries;
+
+  public AsyncUserAccessAuditLogWriter() {
+    mEnabled = Boolean.parseBoolean(Configuration.get(PropertyKey.MASTER_AUDIT_LOGGING_ENABLED));
+    if (mEnabled) {
+      mAuditLogEntries = new ArrayBlockingQueue<>(QUEUE_SIZE);
+    }
+    mStopped = true;
+  }
+
+  public boolean isEnabled() { return mEnabled; }
+
+  public void start() {
+    if (mEnabled && mStopped) {
+      mStopped = false;
+      new Thread(new AuditLoggingWorker()).start();
+    }
+  }
+
+  public void stop() { mStopped = true; }
+
+  public boolean append(AuditContext context) {
```

Comment:
javadoc what the return value means. I did't see any logic to handle the case when `append` returns `false`. do we even bother to return `false`?